src/app/utils/query-url-sanitization.ts

```diff
@@ -1,67 +1,97 @@
+import { GRAPH_URL } from '../services/graph-constants';
 import { parseSampleUrl } from './sample-url-generation';
 
-// Matches patterns like "('<1f7ff346-c174-45e5-af38-294e51d9969a>')" or "('<key>')"
-const keyIdRegex = /\(\'\<?\{?[ ?0-9a-zA-Z-]*\}?\>?\'\)/g;
-
-// Matches patterns like "(query=<key>)" or "(itemat=<key>,mode=<mode>)"
-const functionParamInitialRegex = /(?<=\=).*?(?=,)/g;
-const functionParamFinalRegex = /(?<=\=).[^,]*(?=\))/g;
-
 // Matches pattterns within quotes e.g "displayName: Gupta"
 const quotedTextRegex = /"([^"]*)"/g;
 
-// Matches PII patterns
-const numberRegex =  /(?<=\/)\d+\b/g; // number between forward slashes
-const emailRegex = /([a-z0-9_\-.+]+)@\w+(\.\w+)*/gi;
-const guidRegex =
- /(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}/gi;
-
-
 export function sanitizeQueryUrl(url: string): string {
   url = decodeURIComponent(url);
 
   // Extract query string
-  const { search } = parseSampleUrl(url);
+  const { search, queryVersion, requestUrl } = parseSampleUrl(url);
   const queryString: string = sanitizeQueryParameters(search);
 
-  // Drop query string
-  let sanitizedUrl = url.split('?')[0];
+  let resourceUrl = requestUrl;
+  const sections = resourceUrl.split('/');
+  sections.forEach(sec => {
```

Comment:
I think the logic will be simpler if we check for non-id's and ignore the sanitization process on them.
Non-IDs as of today are all alpha characters, so `^[A-Za-z]+$` could be used to match a full string with alpha characters.
Another possible non-ID format that Michael raised is that the deprecation logic may have `segment_v2` temporarily and we shouldn't sanitize those parts too.

So something like:
```
if (isAllAlpha(sec) || isDeprecation(sec))
{
  // ignore sanitization
}